lib/api/controllers/collectionController.js

```diff
@@ -135,142 +123,132 @@ class CollectionController extends BaseController {
    * Search for collection validation specifications
    *
    * @param {Request} request
-   * @returns {Promise<Object>}
+   * @returns {Promise.<Object>}
    */
-  searchSpecifications(request) {
-    const
-      query = request.input.body && request.input.body.query
-        ? request.input.body.query
-        : {};
-
-    if (request.input.args.size) {
-      const size = request.input.args.size - (request.input.args.from || 0);
+  searchSpecifications (request) {
+    const { from, size, scroll, query } = this.getSearchParams(request);
 
-      if (size > this.kuzzle.config.limits.documentsFetchCount) {
-        errorsManager.throw(
-          'search_page_size',
-          this.kuzzle.config.limits.documentsFetchCount);
-      }
-    }
+    this.assertNotExceedMaxFetch(size, from);
 
-    return this.kuzzle.internalIndex.search('validations', query, request.input.args);
+    return this.kuzzle.internalIndex.search(
+      'validations',
+      query,
+      { from, size, scroll }
+    )
+      .then(({ hits, scrollId, total }) => ({ hits, scrollId, total }));
   }
 
   /**
    * Scroll over a paginated search results
    *
    * @param {Request} request
-   * @returns {Promise<object>}
+   * @returns {Promise.<object>}
    */
-  scrollSpecifications(request) {
-    assertHasScrollId(request);
+  scrollSpecifications (request) {
+    const
+      scrollTTL = this.getNumberParam(request, 'args.scroll', this.defaultScrollTTL),
+      scrollId = this.getStringParam(request, 'args.scrollId');
 
-    return this.kuzzle.internalIndex.scroll(
-      'validations',
-      request.input.args.scrollId,
-      request.input.args.scroll);
+    return this.kuzzle.internalIndex.scroll('validations', scrollId, scrollTTL)
+      .then(({ scrollId, hits, total }) => ({ scrollId, hits, total }));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') "scrollId" hides or potentially hides a variable declared in an outer scope at line 150. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AVariableShadowing)